sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/HCatalogIOTestUtils.java

```diff
@@ -0,0 +1,194 @@
+   * @throws HCatException
+   */
+   static ReaderContext getReaderContext(Map<String, String> config)
+    throws HCatException {
+    ReadEntity entity = new ReadEntity.Builder().withTable(TEST_TABLE_NAME).build();
+    HCatReader reader = DataTransferFactory.getHCatReader(entity, config);
+    ReaderContext context = reader.prepareRead();
+    return context;
+  }
+
+   /**
+    * Returns a WriterContext instance for the passed datastore config params.
+    * @throws HCatException
+    */
+   static WriterContext getWriterContext(Map<String, String> config) throws HCatException {
+      WriteEntity.Builder builder = new WriteEntity.Builder();
+      WriteEntity entity = builder.withTable(TEST_TABLE_NAME).build();
+      HCatWriter writer = DataTransferFactory.getHCatWriter(entity, config);
+      WriterContext context = writer.prepareWrite();
+      return context;
+    }
+
+   private static List<String> readRecords(ReaderContext cntxt, int splitId) throws
+   HCatException {
+     List<String> records = new ArrayList<String>();
+    HCatReader reader = DataTransferFactory.getHCatReader(cntxt, splitId);
+    Iterator<HCatRecord> itr = reader.read();
+    while (itr.hasNext()) {
+      //returning the value of the first column i.e. get(0) as
+      //this should be enough for the testcase
+      records.add(itr.next().get(0).toString());
+    }
+    return records;
+  }
+
+   /**
+    * Writes records to the table using the passed WriterContext.
+    * @throws HCatException
+    */
+   static void writeRecords(WriterContext context) throws HCatException {
+    HCatWriter writer = DataTransferFactory.getHCatWriter(context);
+    writer.write(getHCatRecords(TEST_RECORDS_COUNT).iterator());
+  }
+
+   /**
+    * Commits the pending writes to the database.
+    * @throws IOException
+    */
+   static void commitRecords(Map<String, String> config, WriterContext context)
+       throws IOException {
+    WriteEntity.Builder builder = new WriteEntity.Builder();
+    WriteEntity entity = builder.withTable(TEST_TABLE_NAME).build();
+    HCatWriter writer = DataTransferFactory.getHCatWriter(entity, config);
+    writer.commit(context);
+  }
+
+   /**
+    * Returns a list of strings containing 'expected' test data for verification.
+    */
+   static List<String> getExpectedRecords(int count) {
+     List<String> expected = new ArrayList<String>();
+     for (int i = 0; i < count; i++) {
+       expected.add("record " + i);
+     }
+     return expected;
+   }
+
+   /**
+    * Returns a list of HCatRecords of passed size.
+    */
+   private static List<HCatRecord> getHCatRecords(int size) {
+     List<HCatRecord> expected = new ArrayList<HCatRecord>();
+     for (int i = 0; i < size; i++) {
+       expected.add(toHCatRecord(i));
+     }
+     return expected;
+   }
+
+   /**
+    * Returns a list of DefaultHCatRecords of passed size.
+    */
+   static List<DefaultHCatRecord> getDefaultHCatRecords(int size) {
```

Comment:
Agree with @jkff you can remove this method and rely on getHCatRecords (you should make it accesible) since the transform uses raw HCatRecord. Also you can make the method toHCatRecord return just HCatRecord.